arches/apps.py

```diff
@@ -51,3 +58,62 @@ def check_cache_backend(app_configs, **kwargs):
             )
         )
     return errors
+
+
+@register(Tags.compatibility)
+def check_arches_compatibility(app_configs, **kwargs):
+    try:
+        arches_version = Version(__version__)
+    except ValueError:
+        arches_version = Version.coerce(__version__)
+
+    if app_configs is None:
+        app_configs = apps.get_app_configs()
+
+    errors = []
+    for config in app_configs:
+        if not getattr(config, "is_arches_application", False):
+            continue
+        try:
+            project_requirements = requires(config.name)
+        except PackageNotFoundError:
+            # Not installed by pip: read pyproject.toml directly
+            toml_path = Path(config.path).parent / "pyproject.toml"
+            if not toml_path.exists():
+                raise ValueError
+            with open(toml_path, "rb") as f:
+                data = tomllib.load(f)
+                try:
+                    project_requirements = data["project"]["dependencies"]
+                except KeyError:
+                    raise ValueError from None
+        try:
+            for requirement in project_requirements:
+                if requirement.lower().startswith("arches"):
+                    parsed_arches_requirement = SimpleSpec(
+                        requirement.lower().replace("arches", "").lstrip()
+                    )
+                    break
+            else:
+                raise ValueError
+        except ValueError:
+            errors.append(
+                Error(
+                    f"Invalid or missing arches requirement",
+                    obj=config.name,
+                    hint=project_requirements,
+                    id="arches.E002",
+                )
```

Comment:
Happy to demote this to a warning if that's better. As is, a new project spun up off a version under dev (e.g. 7.6.0.a1) will fail this check (and thus fail to runserver, since any error-level system check will cause manage.py to quit) because a new project is a created with a range of >=7.6.0,<7.7.0 and 7.6.0-a1 is lower than 7.6.0.

We can also adjust the template during the alpha phase as long as we remember to set it to final before releasing final.

Reply:
What about making it a warning if DEBUG==True and an error otherwise?